### ISSUE TITLE
Data type of "sigma"

### ISSUE THREAD

**Issue by cjvanlissa** [cjvanlissa/bain#6]
Sigma is now an atomic vector, covariance matrix, or list of covariance matrices. Would it not be more practical to use an array? List implies that the covariance matrices can have different dimensions - but this seems like something that should not be allowed.

**Comment by xingu123** [comment]
Indeed, covariance matrices should have the same dimension in Bain. I used list because 1. it is easier to use than arrary with which users should understand which dimension is for the number of matrices. To me it is easier to construct a list of matrices rather than a three dimension array. 2. it is easier to handle as we can use length(), lapply() et al. functions on it.

**Comment by cjvanlissa** [comment]
I'm not sure I agree that it is easier for us to use. Array is a simpler datatype than list. With three-dimensional arrays, for example, you can just use apply(x, 3, ...) across the third dimension, and instead of length you can use dim()[3].
For end users, however, you might be right that it's easier to understand. I'll keep this issue open for a bit, because I had a bug which could be traced back to the different data types sigma can have.
